oauth2_http/javatests/com/google/auth/oauth2/ServiceAccountCredentialsTest.java

```diff
@@ -1705,6 +1705,37 @@ public void onFailure(Throwable exception) {
     assertTrue("Should have run onSuccess() callback", success.get());
   }
 
+  @Test
+  public void createScopes_existingAccessTokenInvalidated() throws IOException {
+    PrivateKey privateKey = OAuth2Utils.privateKeyFromPkcs8(PRIVATE_KEY_PKCS8);
+    MockTokenServerTransportFactory transportFactory = new MockTokenServerTransportFactory();
+    transportFactory.transport.addServiceAccount(CLIENT_EMAIL, ACCESS_TOKEN);
+    GoogleCredentials credentials =
+        ServiceAccountCredentials.newBuilder()
+            .setClientId(CLIENT_ID)
+            .setClientEmail(CLIENT_EMAIL)
+            .setPrivateKey(privateKey)
+            .setPrivateKeyId(PRIVATE_KEY_ID)
+            .setProjectId(PROJECT_ID)
+            .setQuotaProjectId("my-quota-project-id")
+            .setHttpTransportFactory(transportFactory)
+            .setUseJwtAccessWithScope(true)
+            .setScopes(SCOPES)
+            .build();
+    // Explicitly call `refresh()` to simulate an API request. Creating a GoogleCredentials object
+    // does not immediately get the Access Token
+    credentials.refresh();
```

Comment:
nit: since you are setting up mock transport, perhaps call `credentials.getRequestMetadata(CALL_URI)`  instead of calling `credentials.refresh()` which user won't need to call directly. Similar calls in tests [here](https://github.com/googleapis/google-auth-library-java/blob/f07d19a505e0ff8c43c3366bcda6350df6dcb3bc/oauth2_http/javatests/com/google/auth/oauth2/ServiceAccountCredentialsTest.java#L635).

Reply:
Thanks, updated.